src/main/java/com/atlassian/db/replica/internal/aurora/AuroraClusterDiscovery.java

```diff
@@ -11,17 +12,21 @@
 import static java.util.stream.Collectors.toList;
 
 public final class AuroraClusterDiscovery {
-    private final AuroraConnectionDetails auroraConnectionDetails;
+    private final ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider;
 
-    public AuroraClusterDiscovery(AuroraConnectionDetails auroraConnectionDetails) {
-        this.auroraConnectionDetails = auroraConnectionDetails;
+    public AuroraClusterDiscovery(ConnectionPoolPerUrlProvider connectionPoolPerUrlProvider) {
+        this.connectionPoolPerUrlProvider = connectionPoolPerUrlProvider;
     }
 
     public Collection<Database> getReplicas(Supplier<Connection> connectionSupplier) {
         try (final Connection connection = connectionSupplier.get()) {
             final AuroraReplicasDiscoverer discoverer = createDiscoverer(connection);
             return discoverer.fetchReplicasUrls(connection).stream()
-                .map(auroraUrl -> new AuroraReplicaNode(auroraUrl, auroraConnectionDetails))
+                .map(auroraUrl -> {
+                    String id = auroraUrl.getEndpoint().getServerId();
+                    ConnectionPool connectionPool = connectionPoolPerUrlProvider.getConnectionPool(auroraUrl);
```

Comment:
Here new connection pool is created every time we want to fetch lsn, connection pool caching will be implemented in separate PR?

Reply:
it's `getConnectionPool` not `createConnectionPool`, it is client's responsibility to reuse existing connection pool